src/main/java/org/antlr/intellij/plugin/ANTLRv4ParserDefinition.java

```diff
@@ -33,13 +36,12 @@ public ANTLRv4ParserDefinition() {
 	@NotNull
 	@Override
 	public Lexer createLexer(Project project) {
-		ANTLRv4Lexer lexer = new ANTLRv4Lexer(null);
-		return new ANTLRv4LexerAdaptor(lexer);
+		return lexer;
```

Comment:
You're sharing a lexer, which has its own state, between every file lexed by the IDE?!